Interest feature applied for every five deposits above $20

okwon2018 opened the pull request:
Changes Made in this PR:
I have implemented feature to incentivize the usage of testudo bank. For every 5th deposit, if the deposit amount is above $20, interest applies.
___
`src/test/java/net/testudobank/helpers`
- Changed `addCustomerToDB(DatabaseDelegate dbDelegate, String ID, String password, String firstName, String lastName, int balance, int interestDeposits)`, so interestDeposits is counted for easier testing
___
`src/main/java/net/testudobank/MvcController.java `
- Created a flag `DEPOSITED` to use in `showLoginForm` and `submitDeposit`. When logging in, it should apply interest to the user.
- Changed `submitDeposit`, so in simple deposit case, the function checks whether the deposit amount is more than $20. If so, increment the `NumDepositsForInterest`, changes `DEPOSITED` to true, and calls `applyInterest` After the call, it sets `DEPOSITED` back to false.
- `applyInterest` is now functioning. It checks if there is no overdraft, balance is not negative, number of deposits is a multiple of 5, and `DEPOSITED` is true.

okwon2018 committed:
de-clutter controller logic with interest helper method

okwon2018 committed:
before switching

okwon2018 committed:
Interest Feature Updated

okwon2018 committed:
Updated Comments for Interest Feature

okwon2018 committed:
Checked decimal amount for the accuracy in

s1ddh4rthc commented:
Hey Ohsung, nice job formatting your PR message. Make sure your commit messages focus on the why and not the what. Nice job making the `DEPOSITED` variable as well as making a helper function for applying the interest rate.

okwon2018 committed:
added 1.5% interest APY per customer incetives and updated frontend to visualize the information to help understand what is going on with their account

okwon2018 commented:
Changed things mentioned in the PR comment
- Changed the commit message
Optionally added a feature
- Added front end feature showing the interest rate when depositing

okwon2018 committed:
added 1.5% interest APY per customer incetives and updated frontend to visualize the information to help understand what is going on with their account

okwon2018 commented:
Deleted additional database which I created to keep track of interest history
